src/main/java/com/google/sps/data/SessionContext.java

```diff
@@ -56,4 +73,21 @@ public String getLoggedInUserId() {
   public boolean isUserLoggedIn() {
     return userService.isUserLoggedIn();
   }
+
+  public boolean isUserExistingInDatastore() throws Exception {
+    String id = getLoggedInUserId();
+    if(id != null) {
+        try {
+            User userExists = PersistentUserRepository.getInstance().fetchUserWithId(id);
+
+            if(userExists.getId().equals(id)){
```

Comment:
nit: The following would simplify the code slightly:
```
return userExists.getId().equals(id);
```

Reply:
Done